sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/ServiceBusJmsConnectionFactoryFactory.java

```diff
@@ -57,12 +60,20 @@ private <T extends ServiceBusJmsConnectionFactory> void setPrefetchPolicy(T fact
     private <T extends ServiceBusJmsConnectionFactory> T createConnectionFactoryInstance(Class<T> factoryClass) {
         try {
             T factory;
-            ServiceBusConnectionString serviceBusConnectionString = new ServiceBusConnectionString(properties.getConnectionString());
-            String host = serviceBusConnectionString.getEndpointUri().getHost();
-
-            String remoteUrl = String.format(AMQP_URI_FORMAT, host, properties.getIdleTimeout().toMillis());
-            String username = serviceBusConnectionString.getSharedAccessKeyName();
-            String password = serviceBusConnectionString.getSharedAccessKey();
+            String remoteUrl = null;
```

Comment:
The `ServiceBusJmsConnectionFactory` shouldn't need to understand the `serviceBusPasswordlessProperties`

Reply:
I have refactored the code, but it's a little strange for me now.

Reply:
`ServiceConnectionStringProvider` is not the best choice.